--- git status ---
HEAD detached from dc9bbf2
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   deploy/config.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/supervised_learning_networks.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	cuncurrent_state_estimator.pth
	tested_policies/go2/go2_cuncurrent_se_with_delayed_start/
	tested_policies/go2/go2_small/model_0.pt
	tested_policies/go2/go2_small/model_100.pt
	tested_policies/go2/go2_small/model_1000.pt
	tested_policies/go2/go2_small/model_1050.pt
	tested_policies/go2/go2_small/model_1100.pt
	tested_policies/go2/go2_small/model_1150.pt
	tested_policies/go2/go2_small/model_1200.pt
	tested_policies/go2/go2_small/model_1250.pt
	tested_policies/go2/go2_small/model_1300.pt
	tested_policies/go2/go2_small/model_1350.pt
	tested_policies/go2/go2_small/model_1400.pt
	tested_policies/go2/go2_small/model_1450.pt
	tested_policies/go2/go2_small/model_150.pt
	tested_policies/go2/go2_small/model_1500.pt
	tested_policies/go2/go2_small/model_1550.pt
	tested_policies/go2/go2_small/model_1600.pt
	tested_policies/go2/go2_small/model_1650.pt
	tested_policies/go2/go2_small/model_1700.pt
	tested_policies/go2/go2_small/model_1750.pt
	tested_policies/go2/go2_small/model_1800.pt
	tested_policies/go2/go2_small/model_1850.pt
	tested_policies/go2/go2_small/model_1900.pt
	tested_policies/go2/go2_small/model_1950.pt
	tested_policies/go2/go2_small/model_200.pt
	tested_policies/go2/go2_small/model_2000.pt
	tested_policies/go2/go2_small/model_2050.pt
	tested_policies/go2/go2_small/model_2100.pt
	tested_policies/go2/go2_small/model_2150.pt
	tested_policies/go2/go2_small/model_2200.pt
	tested_policies/go2/go2_small/model_2250.pt
	tested_policies/go2/go2_small/model_2300.pt
	tested_policies/go2/go2_small/model_2350.pt
	tested_policies/go2/go2_small/model_2400.pt
	tested_policies/go2/go2_small/model_2450.pt
	tested_policies/go2/go2_small/model_250.pt
	tested_policies/go2/go2_small/model_2500.pt
	tested_policies/go2/go2_small/model_2550.pt
	tested_policies/go2/go2_small/model_2600.pt
	tested_policies/go2/go2_small/model_2650.pt
	tested_policies/go2/go2_small/model_2700.pt
	tested_policies/go2/go2_small/model_2750.pt
	tested_policies/go2/go2_small/model_2800.pt
	tested_policies/go2/go2_small/model_2850.pt
	tested_policies/go2/go2_small/model_2900.pt
	tested_policies/go2/go2_small/model_2950.pt
	tested_policies/go2/go2_small/model_300.pt
	tested_policies/go2/go2_small/model_3000.pt
	tested_policies/go2/go2_small/model_3050.pt
	tested_policies/go2/go2_small/model_3100.pt
	tested_policies/go2/go2_small/model_3150.pt
	tested_policies/go2/go2_small/model_3200.pt
	tested_policies/go2/go2_small/model_3250.pt
	tested_policies/go2/go2_small/model_3300.pt
	tested_policies/go2/go2_small/model_3350.pt
	tested_policies/go2/go2_small/model_3400.pt
	tested_policies/go2/go2_small/model_3450.pt
	tested_policies/go2/go2_small/model_350.pt
	tested_policies/go2/go2_small/model_3500.pt
	tested_policies/go2/go2_small/model_3550.pt
	tested_policies/go2/go2_small/model_3600.pt
	tested_policies/go2/go2_small/model_3650.pt
	tested_policies/go2/go2_small/model_3700.pt
	tested_policies/go2/go2_small/model_3750.pt
	tested_policies/go2/go2_small/model_3800.pt
	tested_policies/go2/go2_small/model_3850.pt
	tested_policies/go2/go2_small/model_3900.pt
	tested_policies/go2/go2_small/model_3950.pt
	tested_policies/go2/go2_small/model_400.pt
	tested_policies/go2/go2_small/model_4000.pt
	tested_policies/go2/go2_small/model_4050.pt
	tested_policies/go2/go2_small/model_4100.pt
	tested_policies/go2/go2_small/model_4150.pt
	tested_policies/go2/go2_small/model_4200.pt
	tested_policies/go2/go2_small/model_4250.pt
	tested_policies/go2/go2_small/model_4300.pt
	tested_policies/go2/go2_small/model_4350.pt
	tested_policies/go2/go2_small/model_4400.pt
	tested_policies/go2/go2_small/model_4450.pt
	tested_policies/go2/go2_small/model_450.pt
	tested_policies/go2/go2_small/model_4500.pt
	tested_policies/go2/go2_small/model_4550.pt
	tested_policies/go2/go2_small/model_4600.pt
	tested_policies/go2/go2_small/model_4650.pt
	tested_policies/go2/go2_small/model_4700.pt
	tested_policies/go2/go2_small/model_4750.pt
	tested_policies/go2/go2_small/model_4800.pt
	tested_policies/go2/go2_small/model_4850.pt
	tested_policies/go2/go2_small/model_4900.pt
	tested_policies/go2/go2_small/model_4950.pt
	tested_policies/go2/go2_small/model_50.pt
	tested_policies/go2/go2_small/model_500.pt
	tested_policies/go2/go2_small/model_5000.pt
	tested_policies/go2/go2_small/model_5050.pt
	tested_policies/go2/go2_small/model_5100.pt
	tested_policies/go2/go2_small/model_5150.pt
	tested_policies/go2/go2_small/model_5200.pt
	tested_policies/go2/go2_small/model_5250.pt
	tested_policies/go2/go2_small/model_5300.pt
	tested_policies/go2/go2_small/model_5350.pt
	tested_policies/go2/go2_small/model_5400.pt
	tested_policies/go2/go2_small/model_5450.pt
	tested_policies/go2/go2_small/model_550.pt
	tested_policies/go2/go2_small/model_5500.pt
	tested_policies/go2/go2_small/model_5550.pt
	tested_policies/go2/go2_small/model_5600.pt
	tested_policies/go2/go2_small/model_5650.pt
	tested_policies/go2/go2_small/model_5700.pt
	tested_policies/go2/go2_small/model_5750.pt
	tested_policies/go2/go2_small/model_5800.pt
	tested_policies/go2/go2_small/model_5850.pt
	tested_policies/go2/go2_small/model_5900.pt
	tested_policies/go2/go2_small/model_5950.pt
	tested_policies/go2/go2_small/model_600.pt
	tested_policies/go2/go2_small/model_6000.pt
	tested_policies/go2/go2_small/model_6050.pt
	tested_policies/go2/go2_small/model_6100.pt
	tested_policies/go2/go2_small/model_6150.pt
	tested_policies/go2/go2_small/model_6200.pt
	tested_policies/go2/go2_small/model_6250.pt
	tested_policies/go2/go2_small/model_6300.pt
	tested_policies/go2/go2_small/model_6350.pt
	tested_policies/go2/go2_small/model_6400.pt
	tested_policies/go2/go2_small/model_6450.pt
	tested_policies/go2/go2_small/model_650.pt
	tested_policies/go2/go2_small/model_6500.pt
	tested_policies/go2/go2_small/model_6550.pt
	tested_policies/go2/go2_small/model_6600.pt
	tested_policies/go2/go2_small/model_6650.pt
	tested_policies/go2/go2_small/model_6700.pt
	tested_policies/go2/go2_small/model_6750.pt
	tested_policies/go2/go2_small/model_6800.pt
	tested_policies/go2/go2_small/model_6850.pt
	tested_policies/go2/go2_small/model_6900.pt
	tested_policies/go2/go2_small/model_6950.pt
	tested_policies/go2/go2_small/model_700.pt
	tested_policies/go2/go2_small/model_7000.pt
	tested_policies/go2/go2_small/model_7050.pt
	tested_policies/go2/go2_small/model_7100.pt
	tested_policies/go2/go2_small/model_7150.pt
	tested_policies/go2/go2_small/model_7200.pt
	tested_policies/go2/go2_small/model_7250.pt
	tested_policies/go2/go2_small/model_7300.pt
	tested_policies/go2/go2_small/model_7350.pt
	tested_policies/go2/go2_small/model_7400.pt
	tested_policies/go2/go2_small/model_7450.pt
	tested_policies/go2/go2_small/model_750.pt
	tested_policies/go2/go2_small/model_7500.pt
	tested_policies/go2/go2_small/model_7550.pt
	tested_policies/go2/go2_small/model_7600.pt
	tested_policies/go2/go2_small/model_7650.pt
	tested_policies/go2/go2_small/model_7700.pt
	tested_policies/go2/go2_small/model_7750.pt
	tested_policies/go2/go2_small/model_7800.pt
	tested_policies/go2/go2_small/model_7850.pt
	tested_policies/go2/go2_small/model_7900.pt
	tested_policies/go2/go2_small/model_7950.pt
	tested_policies/go2/go2_small/model_800.pt
	tested_policies/go2/go2_small/model_850.pt
	tested_policies/go2/go2_small/model_900.pt
	tested_policies/go2/go2_small/model_950.pt

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/deploy/config.py b/deploy/config.py
index f2e7540..3f4f238 100644
--- a/deploy/config.py
+++ b/deploy/config.py
@@ -4,7 +4,7 @@ dir_path = os.path.dirname(os.path.realpath(__file__))
 sys.path.append(dir_path+"/../")
 sys.path.append(dir_path+"/../scripts/rsl_rl")
 
-robot = 'aliengo'  # 'aliengo', 'go1', 'go2', 'b2', 'hyqreal1', 'hyqreal2', 'mini_cheetah' 
+robot = 'go2'  # 'aliengo', 'go1', 'go2', 'b2', 'hyqreal1', 'hyqreal2', 'mini_cheetah' 
 scene = 'random_boxes'  # flat, random_boxes, random_pyramids, perlin
 
 # ----------------------------------------------------------------------------------------------------------------
@@ -38,7 +38,7 @@ else:
 
 # ----------------------------------------------------------------------------------------------------------------
 
-policy_folder_path = dir_path + "/../tested_policies/" + robot + "/aliengo_symmetricactor"
+policy_folder_path = dir_path + "/../tested_policies/" + robot + "/go2_cuncurrent_se_flat"
 #policy_folder_path = dir_path + "/../tested_policies/" + robot + "/go2_5asymm"
 
 cuncurrent_state_est_network = policy_folder_path + "/exported/cuncurrent_state_estimator.pth"
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
index bd64fa6..ad858ef 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
@@ -16,9 +16,9 @@ damping_mujoco = 2.0
 
 armature_mujoco = 0.01
 
-friction_static_mujoco = 0.2# * 0.0
-friction_dynamic_mujoco = 0.6# * 0.0
-activation_vel = 0.1# * 0.0
+friction_static_mujoco = 0.1# * 0.0
+friction_dynamic_mujoco = 0.0# * 0.0
+activation_vel = 0.01# * 0.0
 
 static_friction_hip = 0.5 * 0.0
 dynamic_friction_hip = 0.3 * 0.0
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
index cb55a8b..8fc520c 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
@@ -139,7 +139,7 @@ class RoughPPORunnerCfg(RslRlOnPolicyRunnerCfg):
         activation="elu",
     )
     algorithm = RslRlPpoAlgorithmCfg(
-        class_name="PPO", #PPO, PPOSymmDataAugmented #AMP_PPO
+        class_name="PPOSymmDataAugmented", #PPO, PPOSymmDataAugmented #AMP_PPO
         value_loss_coef=1.0,
         use_clipped_value_loss=True,
         clip_param=0.2,
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py
index 906ac5e..db7311e 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/go2_env_cfg.py
@@ -166,8 +166,8 @@ class Go2FlatEnvCfg(DirectRLEnvCfg):
         cuncurrent_state_est_batch_size = 512
         cuncurrent_state_est_train_epochs = 1000
         cuncurrent_state_est_lr = 1e-3
-        cuncurrent_state_est_ep_saving_interval = 1000
-        cuncurrent_state_est_ep_saving_start = 6000 
+        cuncurrent_state_est_ep_saving_interval = 800
+        cuncurrent_state_est_ep_saving_start = 800 
 
     use_rma = False
     if(use_rma):
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/supervised_learning_networks.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/supervised_learning_networks.py
index 5d97c40..3206c8f 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/supervised_learning_networks.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/supervised_learning_networks.py
@@ -13,7 +13,7 @@ class CustomDataset(Dataset):
         # There is a problem! we append (num_envs, features) as one element inside the list, not N!
 
         # Save only random 128 element from the input_data and label
-        random_idx = random.sample(range(input_data.size(0)), min(128, input_data.size(0)))
+        random_idx = random.sample(range(input_data.size(0)), min(512, input_data.size(0)))
         input_data_cpu = input_data[random_idx].clone().detach().cpu()
         label_cpu = label[random_idx].clone().detach().cpu()
 
@@ -88,7 +88,7 @@ class SimpleNN(torch.nn.Module):
     def forward(self, x):
         x = torch.relu(self.fc1(x))
         x = torch.relu(self.fc2(x))
-        x = self.fc3(x)
+        x = torch.tanh(self.fc3(x))*2.0
         return x
     
 