--- git status ---
HEAD detached from dc9bbf2
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   deploy/config.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_cfg.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_electric.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_hydraulic.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/b2_asset.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/hyqreal_asset.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/aliengo_env_cfg.py
	modified:   source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/locomotion_env.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	cuncurrent_state_estimator.pth

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/deploy/config.py b/deploy/config.py
index f2e7540..a71bfbb 100644
--- a/deploy/config.py
+++ b/deploy/config.py
@@ -4,7 +4,7 @@ dir_path = os.path.dirname(os.path.realpath(__file__))
 sys.path.append(dir_path+"/../")
 sys.path.append(dir_path+"/../scripts/rsl_rl")
 
-robot = 'aliengo'  # 'aliengo', 'go1', 'go2', 'b2', 'hyqreal1', 'hyqreal2', 'mini_cheetah' 
+robot = 'go2'  # 'aliengo', 'go1', 'go2', 'b2', 'hyqreal1', 'hyqreal2', 'mini_cheetah' 
 scene = 'random_boxes'  # flat, random_boxes, random_pyramids, perlin
 
 # ----------------------------------------------------------------------------------------------------------------
@@ -38,7 +38,7 @@ else:
 
 # ----------------------------------------------------------------------------------------------------------------
 
-policy_folder_path = dir_path + "/../tested_policies/" + robot + "/aliengo_symmetricactor"
+policy_folder_path = dir_path + "/../tested_policies/" + robot + "/delay_rough_with_cuncurrent_se"
 #policy_folder_path = dir_path + "/../tested_policies/" + robot + "/go2_5asymm"
 
 cuncurrent_state_est_network = policy_folder_path + "/exported/cuncurrent_state_estimator.pth"
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_cfg.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_cfg.py
index 6de4ca2..931a54a 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_cfg.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_cfg.py
@@ -6,14 +6,14 @@
 from dataclasses import MISSING
 
 from isaaclab.utils import configclass
-from isaaclab.actuators import DCMotorCfg
+from isaaclab.actuators import DCMotorCfg, DelayedPDActuatorCfg
 
 from .actuator_electric import IdentifiedActuatorElectric
 from .actuator_hydraulic import IdentifiedActuatorHydraulic
 
 
 @configclass
-class IdentifiedActuatorElectricCfg(DCMotorCfg):
+class IdentifiedActuatorElectricCfg(DelayedPDActuatorCfg):
     """Configuration for direct control (DC) motor actuator model."""
 
     class_type: type = IdentifiedActuatorElectric
@@ -32,7 +32,7 @@ class IdentifiedActuatorElectricCfg(DCMotorCfg):
     """ (in percentage)."""
 
 @configclass
-class IdentifiedActuatorHydraulicCfg(DCMotorCfg):
+class IdentifiedActuatorHydraulicCfg(DelayedPDActuatorCfg):
     """Configuration for direct control (DC) motor actuator model."""
 
     class_type: type = IdentifiedActuatorHydraulic
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_electric.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_electric.py
index 2266c18..2746ca1 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_electric.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_electric.py
@@ -10,14 +10,14 @@ from typing import TYPE_CHECKING
 
 from isaacsim.core.utils.types import ArticulationActions
 
-from isaaclab.actuators import DCMotor
+from isaaclab.actuators import DCMotor, DelayedPDActuator
 from isaaclab.utils import DelayBuffer, LinearInterpolation
 
 if TYPE_CHECKING:
     from .actuator_cfg import IdentifiedActuatorElectricCfg
 
 
-class IdentifiedActuatorElectric(DCMotor):
+class IdentifiedActuatorElectric(DelayedPDActuator):
     cfg: IdentifiedActuatorElectricCfg
 
     def __init__(self, cfg: IdentifiedActuatorElectricCfg, *args, **kwargs):
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_hydraulic.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_hydraulic.py
index 4c97758..aadbc0b 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_hydraulic.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/actuators/actuator_hydraulic.py
@@ -10,14 +10,14 @@ from typing import TYPE_CHECKING
 
 from isaacsim.core.utils.types import ArticulationActions
 
-from isaaclab.actuators import DCMotor
+from isaaclab.actuators import DCMotor, DelayedPDActuator
 from isaaclab.utils import DelayBuffer, LinearInterpolation
 
 if TYPE_CHECKING:
     from .actuator_cfg import IdentifiedActuatorHydraulicCfg
 
 
-class IdentifiedActuatorHydraulic(DCMotor):
+class IdentifiedActuatorHydraulic(DelayedPDActuator):
     cfg: IdentifiedActuatorHydraulicCfg
 
     def __init__(self, cfg: IdentifiedActuatorHydraulicCfg, *args, **kwargs):
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset.py
index 8edc787..5f10c22 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/aliengo_asset.py
@@ -33,7 +33,7 @@ ALIENGO_HIP_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     joint_names_expr=[".*_hip_joint"],
     effort_limit=44.4,
     velocity_limit=21.0,
-    saturation_effort=44.4,
+    #saturation_effort=44.4,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
@@ -44,13 +44,16 @@ ALIENGO_HIP_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     friction = static_friction_hip,
     dynamic_friction = dynamic_friction_hip,
     viscous_friction = viscous_friction_hip,
+
+    min_delay=1,  # physics time steps (min: 2.0*0=0.0ms)
+    max_delay=4,  # physics time steps (max: 2.0*4=8.0ms)
 )
 
 ALIENGO_THIGH_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     joint_names_expr=[".*_thigh_joint"],
     effort_limit=44.4,
     velocity_limit=21.0,
-    saturation_effort=44.4,
+    #saturation_effort=44.4,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
@@ -61,13 +64,16 @@ ALIENGO_THIGH_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     friction = static_friction_thigh,
     dynamic_friction = dynamic_friction_thigh,
     viscous_friction = viscous_friction_thigh,
+
+    min_delay=1,  # physics time steps (min: 2.0*0=0.0ms)
+    max_delay=4,  # physics time steps (max: 2.0*4=8.0ms)
 )
 
 ALIENGO_CALF_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     joint_names_expr=[".*_calf_joint"],
     effort_limit=44.4,
     velocity_limit=21.0,
-    saturation_effort=44.4,
+    #saturation_effort=44.4,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
@@ -78,6 +84,9 @@ ALIENGO_CALF_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     friction = static_friction_calf,
     dynamic_friction = dynamic_friction_calf,
     viscous_friction = viscous_friction_calf,
+
+    min_delay=1,  # physics time steps (min: 2.0*0=0.0ms)
+    max_delay=4,  # physics time steps (max: 2.0*4=8.0ms)
 )
 
 
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/b2_asset.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/b2_asset.py
index 17733f6..4505dc4 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/b2_asset.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/b2_asset.py
@@ -21,39 +21,48 @@ B2_HIP_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     joint_names_expr=[".*_hip_joint"],
     effort_limit=44.4,
     velocity_limit=21.0,
-    saturation_effort=44.4,
+    #saturation_effort=44.4,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
     friction_static=friction_static_mujoco,
     activation_vel=0.1,
     friction_dynamic=friction_dynamic_mujoco,
+
+    min_delay=0,  # physics time steps (min: 2.0*0=0.0ms)
+    max_delay=2,  # physics time steps (max: 2.0*4=8.0ms)
 )
 
 B2_THIGH_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     joint_names_expr=[".*_thigh_joint"],
     effort_limit=44.4,
     velocity_limit=21.0,
-    saturation_effort=44.4,
+    #saturation_effort=44.4,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
     friction_static=friction_static_mujoco,
     activation_vel=0.1,
     friction_dynamic=friction_dynamic_mujoco,
+
+    min_delay=0,  # physics time steps (min: 2.0*0=0.0ms)
+    max_delay=2,  # physics time steps (max: 2.0*4=8.0ms)
 )
 
 B2_CALF_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     joint_names_expr=[".*_calf_joint"],
     effort_limit=44.4,
     velocity_limit=21.0,
-    saturation_effort=44.4,
+    #saturation_effort=44.4,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
     friction_static=friction_static_mujoco,
     activation_vel=0.1,
     friction_dynamic=friction_dynamic_mujoco,
+
+    min_delay=0,  # physics time steps (min: 2.0*0=0.0ms)
+    max_delay=2,  # physics time steps (max: 2.0*4=8.0ms)
 )
 
 
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
index 4f7b7a1..927b16a 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/go2_asset.py
@@ -36,7 +36,7 @@ GO2_HIP_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     joint_names_expr=[".*_hip_joint"],
     effort_limit=23.7,
     velocity_limit=30.1,
-    saturation_effort=23.7,
+    #saturation_effort=23.7,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
@@ -47,13 +47,16 @@ GO2_HIP_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     friction = static_friction_hip,
     dynamic_friction = dynamic_friction_hip,
     viscous_friction = viscous_friction_hip,
+
+    min_delay=1,  # physics time steps (min: 2.0*0=0.0ms)
+    max_delay=4,  # physics time steps (max: 2.0*4=8.0ms)
 )
 
 GO2_THIGH_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     joint_names_expr=[".*_thigh_joint"],
     effort_limit=23.7,
     velocity_limit=30.1,
-    saturation_effort=23.7,
+    #saturation_effort=23.7,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
@@ -64,13 +67,16 @@ GO2_THIGH_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     friction = static_friction_thigh,
     dynamic_friction = dynamic_friction_thigh,
     viscous_friction = viscous_friction_thigh,
+
+    min_delay=1,  # physics time steps (min: 2.0*0=0.0ms)
+    max_delay=4,  # physics time steps (max: 2.0*4=8.0ms)
 )
 
 GO2_CALF_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     joint_names_expr=[".*_calf_joint"],
     effort_limit=45.43,
     velocity_limit=15.7,
-    saturation_effort=45.43,
+    #saturation_effort=45.43,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
@@ -81,6 +87,9 @@ GO2_CALF_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
     friction = static_friction_calf,
     dynamic_friction = dynamic_friction_calf,
     viscous_friction = viscous_friction_calf,
+
+    min_delay=1,  # physics time steps (min: 2.0*0=0.0ms)
+    max_delay=4,  # physics time steps (max: 2.0*4=8.0ms)
 )
 
 
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/hyqreal_asset.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/hyqreal_asset.py
index 3a8b212..cc49da0 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/hyqreal_asset.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/assets/hyqreal_asset.py
@@ -37,7 +37,7 @@ HYQREAL_HIP_ACTUATOR_CFG = IdentifiedActuatorHydraulicCfg(
     joint_names_expr=[".*_hip_joint"],
     effort_limit=173.0,
     velocity_limit=21.0,
-    saturation_effort=173.0,
+    #saturation_effort=173.0,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
@@ -50,13 +50,16 @@ HYQREAL_HIP_ACTUATOR_CFG = IdentifiedActuatorHydraulicCfg(
     friction = static_friction_hip,
     dynamic_friction = dynamic_friction_hip,
     viscous_friction = viscous_friction_hip,
+
+    min_delay=0,  # physics time steps (min: 2.0*0=0.0ms)
+    max_delay=2,  # physics time steps (max: 2.0*4=8.0ms)
 )
 
 HYQREAL_THIGH_ACTUATOR_CFG = IdentifiedActuatorHydraulicCfg(
     joint_names_expr=[".*_thigh_joint"],
     effort_limit=208.0,
     velocity_limit=21.0,
-    saturation_effort=208.0,
+    #saturation_effort=208.0,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
@@ -69,13 +72,16 @@ HYQREAL_THIGH_ACTUATOR_CFG = IdentifiedActuatorHydraulicCfg(
     friction = static_friction_thigh,
     dynamic_friction = dynamic_friction_thigh,
     viscous_friction = viscous_friction_thigh,
+
+    min_delay=0,  # physics time steps (min: 2.0*0=0.0ms)
+    max_delay=2,  # physics time steps (max: 2.0*4=8.0ms)
 )
 
 HYQREAL_CALF_ACTUATOR_CFG = IdentifiedActuatorHydraulicCfg(
     joint_names_expr=[".*_calf_joint"],
     effort_limit=249.0,
     velocity_limit=21.0,
-    saturation_effort=249.0,
+    #saturation_effort=249.0,
     stiffness=stiffness_mujoco,
     damping=damping_mujoco,
     armature=armature_mujoco,
@@ -88,6 +94,9 @@ HYQREAL_CALF_ACTUATOR_CFG = IdentifiedActuatorHydraulicCfg(
     friction = static_friction_calf,
     dynamic_friction = dynamic_friction_calf,
     viscous_friction = viscous_friction_calf,
+
+    min_delay=0,  # physics time steps (min: 2.0*0=0.0ms)
+    max_delay=2,  # physics time steps (max: 2.0*4=8.0ms)
 )
 
 
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
index 8fc520c..cb55a8b 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/agents/rsl_rl_ppo_cfg.py
@@ -139,7 +139,7 @@ class RoughPPORunnerCfg(RslRlOnPolicyRunnerCfg):
         activation="elu",
     )
     algorithm = RslRlPpoAlgorithmCfg(
-        class_name="PPOSymmDataAugmented", #PPO, PPOSymmDataAugmented #AMP_PPO
+        class_name="PPO", #PPO, PPOSymmDataAugmented #AMP_PPO
         value_loss_coef=1.0,
         use_clipped_value_loss=True,
         clip_param=0.2,
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/aliengo_env_cfg.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/aliengo_env_cfg.py
index 5cf3262..84543a7 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/aliengo_env_cfg.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/aliengo_env_cfg.py
@@ -159,7 +159,7 @@ class AliengoFlatEnvCfg(DirectRLEnvCfg):
 
     use_imu = False
 
-    use_cuncurrent_state_est = False
+    use_cuncurrent_state_est = True
     if(use_cuncurrent_state_est):
         cuncurrent_state_est_output_space = 3 #lin_vel_b
         single_cuncurrent_state_est_observation_space = single_observation_space
@@ -254,7 +254,12 @@ class AliengoFlatEnvCfg(DirectRLEnvCfg):
     )
 
     # an imu sensor in case we don't want any state estimator
-    imu = ImuCfg(prim_path="/World/envs/env_.*/Robot/base", debug_vis=True)
+    imu = ImuCfg(
+        prim_path="/World/envs/env_.*/Robot/base", 
+        offset=ImuCfg.OffsetCfg(
+            pos=(-0.02557, 0, 0.04232)
+        ), 
+        debug_vis=False)
 
 
     # scene
diff --git a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/locomotion_env.py b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/locomotion_env.py
index 7903c9a..af6ad74 100644
--- a/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/locomotion_env.py
+++ b/source/basic_locomotion_dls_isaaclab/basic_locomotion_dls_isaaclab/tasks/locomotion/locomotion_env.py
@@ -660,19 +660,22 @@ class LocomotionEnv(DirectRLEnv):
         self._commands[:, :3] = self._commands[:, :3] * ~resample_time.unsqueeze(1).expand(-1, 3) + commands_resample * resample_time.unsqueeze(1).expand(-1, 3)
 
         # Stop
-        rest_time = torch.logical_and(
+        """rest_time = torch.logical_and(
             self.episode_length_buf >= self.max_episode_length - 250,
             self.episode_length_buf < self.max_episode_length - 150
-        )
+        )"""
+        rest_time = self.episode_length_buf >= self.max_episode_length - 50
         self._commands[:, :3] *= ~rest_time.unsqueeze(1).expand(-1, 3)
 
-        # Move again
+        self._commands[:, 3:] *= 0.0
+
+        """# Move again
         resample_time_2 = self.episode_length_buf == self.max_episode_length - 150
         commands_resample_2 = torch.zeros_like(self._commands).uniform_(-1.0, 1.0)
         commands_resample_2[:, 0] *= 0.5
         commands_resample_2[:, 1] *= 0.25 
         commands_resample_2[:, 2] *= 0.5 
-        self._commands[:, :3] = self._commands[:, :3] * ~resample_time_2.unsqueeze(1).expand(-1, 3) + commands_resample_2 * resample_time_2.unsqueeze(1).expand(-1, 3)        
+        self._commands[:, :3] = self._commands[:, :3] * ~resample_time_2.unsqueeze(1).expand(-1, 3) + commands_resample_2 * resample_time_2.unsqueeze(1).expand(-1, 3)"""
 
         # Took some envs, and put to zero the vel
         if self.num_envs > 100: